### DISCUSSION TITLE
[Exercise-10] Write a few words about primary key

### DISCUSSION THREAD

**Discussion by acrlakshman** [exaslate-courses/oracle-sql-plsql-b3#11] [General]
1. What is a primary key?
2. Can you create a primary key with more than one column in a table?
3. Can you create a primary key after table is created?
4. Maximum number of columns that can be used in a primary key?
5. How many columns in a primary key are allowed to be `null`?

**Comment by cm2024jsp** [comment]
1. The PRIMARY KEY constraint uniquely identifies each record in a table. Primary keys must contain UNIQUE values, and cannot contain NULL values.
2. NO
3. NO
4. 1 Primary key
5. 0

**Reply by acrlakshman** [reply]
@cm2024jsp thank you for posting the comments Sai. responses to 2,3,4 are not correct.

2. More than one column can be used to create a primary key.
3. Yes we can create primary key using `ALTER` after creating a table.
4. Oracle allows 32 columns to create a primary key.

**Comment by cm2024jsp** [comment]
1.The PRIMARY KEY constraint uniquely identifies each record in a table. Primary keys must contain UNIQUE values, and cannot contain NULL values.
2.More than one column can be used to create a primary key.
3.Yes we can create primary key using ALTER after creating a table.
4.Oracle allows 32 columns to create a primary key.
5.0

**Comment by srideviadapa0511** [comment]
Primary key is a combination of one more columns in a table.It consists of a unique constraint. A primary key  must be unique and there should be no redundancy or duplication is not allowed .A primary key shouldn't be changed and the value shouldn't be null.
2.Yes we can use more than one column in creating foreign key.
3.Yes we can create primary key after table creation by using a keyword Alter WHICH IS A DDL statement.
4. oracle allows 32 columns to create a primary key .
5. Primary key should always have some value and shouldn't left null.

**Reply by acrlakshman** [reply]
@srideviadapa0511 can you also answer all the questions I asked above?

**Reply by srideviadapa0511** [reply]
sir i edited my comment now . please go through.

**Reply by acrlakshman** [reply]
Nice work. @srideviadapa0511

**Comment by YaminiYadavv** [comment]
1. What is a primary key?
Ans: A Primary key is a combination of one or more columns in a table, they cannot be changed and remains unique constraints in nature and even does not allow any duplicates to create.
 2. Can you create a primary key with more than one column in a table?
 Ans: Yes, we can create more than one.
3. Can you create a primary key after table is created?
Ans: yes we can create using 'alter' keyword.
4. Maximum number of columns that can be used in a primary key?
Ans:  32
5. How many columns in a primary key are allowed to be null?
Ans: No, It cannot be null as Primary key must be always assigned with a value.
